projects/organization-management/src/app/components/user-profile-form/user-profile-form.component.ts

```diff
@@ -1,41 +1,146 @@
-import { ChangeDetectionStrategy, Component, Input, OnDestroy, OnInit } from '@angular/core';
+import { ChangeDetectionStrategy, Component, Input, OnInit } from '@angular/core';
 import { FormGroup } from '@angular/forms';
-import { Observable, Subject } from 'rxjs';
-import { takeUntil } from 'rxjs/operators';
+import { FormlyFieldConfig } from '@ngx-formly/core';
+import { TranslateService } from '@ngx-translate/core';
+import { pick } from 'lodash-es';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
 
 import { AppFacade } from 'ish-core/facades/app.facade';
 import { HttpError } from 'ish-core/models/http-error/http-error.model';
-import { Locale } from 'ish-core/models/locale/locale.model';
 import { whenTruthy } from 'ish-core/utils/operators';
+import { SelectOption } from 'ish-shared/forms/components/select/select.component';
 import { determineSalutations } from 'ish-shared/forms/utils/form-utils';
+import { SpecialValidators } from 'ish-shared/forms/validators/special-validators';
+
+import { B2bUser } from '../../models/b2b-user/b2b-user.model';
 
 @Component({
   selector: 'ish-user-profile-form',
   templateUrl: './user-profile-form.component.html',
   changeDetection: ChangeDetectionStrategy.Default,
 })
-export class UserProfileFormComponent implements OnInit, OnDestroy {
+export class UserProfileFormComponent implements OnInit {
   @Input() form: FormGroup;
   @Input() error: HttpError;
+  @Input() user?: B2bUser;
 
-  currentLocale$: Observable<Locale>;
-  private destroy$ = new Subject();
+  fields: FormlyFieldConfig[];
+  model: Partial<B2bUser>;
 
-  titles = [];
+  titleOptions$: Observable<SelectOption[]>;
 
-  constructor(private appFacade: AppFacade) {}
+  constructor(private appFacade: AppFacade, private translate: TranslateService) {}
 
   ngOnInit() {
-    this.currentLocale$ = this.appFacade.currentLocale$;
-
     // determine default language from session and available locales
-    this.currentLocale$.pipe(whenTruthy(), takeUntil(this.destroy$)).subscribe(locale => {
-      this.titles = locale?.lang ? determineSalutations(locale.lang.slice(3)) : undefined;
-    });
+    this.titleOptions$ = this.appFacade.currentLocale$.pipe(
+      whenTruthy(),
+      map(locale =>
+        determineSalutations(locale.lang?.slice(3)).map(title => ({
+          value: this.translate.instant(title),
+          label: title,
+        }))
+      )
+    );
+
+    this.model = this.setModel(this.user);
+    this.fields = this.setFields();
```

Comment:
This is definitely small but if the method returns fields, shouldn't it be called getFields() and getModel() ?